polar_route/crossing_smoothing.py

```diff
@@ -0,0 +1,1546 @@
+        elif abs(case) == 2:
+            # Defining the min and max of the start and end cells
+            smin = cell_a['cy']-cell_a['dcy'] 
+            smax = cell_a['cy']+cell_a['dcy']
+            emin = cell_b['cy']-cell_b['dcy']
+            emax = cell_b['cy']+cell_b['dcy']
+
+            # Defining the global min and max
+            vmin = np.max([smin,emin])
+            vmax = np.min([smax,emax])
+
+            # Point crossingpoint on boundary between the two origional cells
+            if (x[1] >= vmin) and (x[1] <= vmax):
+                return None,None
+
+            # If Start and end cells share a edge for the horesshoe 
+            if (x[1]<smin) and (smin==emin):
+                case_a = 4
+                case_b   = 4
+            if (x[1]>smax) and (smax==emax):
+                case_a = -4
+                case_b   = -4
+
+            # --- Cases where StartCell is Larger than end Cell ---
+            if (x[1]>emax) and (smax>emax):
+                case_a = case
+                case_b   = (-4)                
+            if (x[1]<emin) and (smin<emin):
+                case_a = case
+                case_b   = (4)                   
+
+            # --- Cases where StartCell is smaller than end Cell ---
+            if (x[1]>smax) and (smax<emax):
+                case_a = -4
+                case_b   = -case
+            if (x[1]<smin) and (emin<smin):
+                case_a = 4
+                case_b   = -case      
+
+        elif abs(case) == 4:
+            # Defining the min and max of the start and end cells
+            smin = cell_a['cx']-cell_a['dcx']
+            smax = cell_a['cx']+cell_a['dcx']
+            emin = cell_b['cx']-cell_b['dcx']
+            emax = cell_b['cx']+cell_b['dcx']
+
+            # Defining the global min and max
+            vmin = np.max([smin,emin])
+            vmax = np.min([smax,emax])
+
+            # Point crossingpoint on boundary between the two origional cells
+            if (x[0] >= vmin) and (x[0] <= vmax):
+                return None,None
+
+            # If Start and end cells share a edge for the horesshoe 
+            if (x[0]<smin) and (smin==emin):
+                case_a = -2
+                case_b   = -2
+            if (x[0]>smax) and (smax==emax):
+                case_a = 2
+                case_b   = 2
+
+            # --- Cases where StartCell is Larger than end Cell ---
+            if (x[0]>emax) and (smax>emax):
+                case_a = case
+                case_b   = (2)                
+            if (x[0]<emin) and (smin<emin):
+                case_a = case
+                case_b   = (-2)                   
+
+            # --- Cases where StartCell is smaller than end Cell ---
+            if (x[0]>smax) and (smax<emax):
+                case_a = (2)
+                case_b   = -case
+            if (x[0]<smin) and (emin<smin):
+                case_a = (-2)
+                case_b   = -case   
+
+        try:
```

Comment:
Does this need to be wrapped in a try/except? When would returning these variables raise an exception? If it's just when one is undefined you could set a default value of `None` either before the previous logic or as part of an `else` statement somewhere.

Reply:
This is a good point. I have reformatted the code and now return a raise Exception as this function should never get to the end without failure